CMakeLists.txt

```
@@ -0,0 +1,193 @@
+cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
+
+include(CMakeDependentOption)
+include(CheckSymbolExists)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_POSITION_INDEPENDENT_CODE YES)
+
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+option(ENABLE_DEBUG_POSTFIX "Add \"d\" after library names for debug builds" ON)
+if(ENABLE_DEBUG_POSTFIX)
+    set(CMAKE_DEBUG_POSTFIX d)
+endif()
+
+# The API (SO) and detailed library versions for the shared library.
+set(PROJECTM_SO_VERSION "3")
+set(PROJECTM_LIB_VERSION "3.1.1")
+
+project(projectm
+        LANGUAGES C CXX
+        VERSION 3.1.13
+        )
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+set(PROJECTM_BIN_DIR "bin" CACHE STRING "Executable installation directory, relative to the install prefix.")
+set(PROJECTM_DATADIR_PATH "share/projectM" CACHE STRING "Default (absolute) path to the projectM data files (presets etc.)")
+set(PROJECTM_LIB_DIR "lib" CACHE STRING "Library installation directory, relative to the install prefix.")
+set(PROJECTM_INCLUDE_DIR "include" CACHE STRING "Header installation directory, relative to the install prefix.")
+
+# Feature options, including dependencies.
+option(ENABLE_PRESETS "Build and install bundled presets" ON)
```

```suggestion
option(PRESETS "Build and install bundled presets" ON)
```

This `ENABLE_` prefix for options is unconventional.